### DISCUSSION TITLE
esp32 target -> host folder and provider.js

### DISCUSSION THREAD

**Discussion by tribalmatt** [Moddable-OpenSource/moddable#718] [General]
Hi guys,

As I clean up my heltec esp32 target.  What is the purpose of provider.js in the hosts folder.  I did not find a reference in the readme. 

Thx,
Matt

**Comment by phoddie** [comment]
We are in the process of migrating from the current Moddable SDK `pins` modules for IO to Ecma-419 based IO. The `provider.js` module is a key part of that. It implements the [Host Provider Instance](https://419.ecma-international.org/#-16-host-provider-instance)  from Ecma-419, ECMAScript® Embedded Systems API Specification.
